apps/emqx_gateway/test/emqx_coap_SUITE.erl

```diff
@@ -0,0 +1,289 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2020-2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_coap_SUITE).
+
+-compile(export_all).
+-compile(nowarn_export_all).
+
+-include_lib("er_coap_client/include/coap.hrl").
+-include_lib("emqx/include/emqx.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("common_test/include/ct.hrl").
+
+-define(CONF_DEFAULT, <<"
+gateway.coap {
+              idle_timeout = 30s
+              enable_stats = false
+              mountpoint = \"\"
+              notify_type = qos
+              connection_required = true
+              subscribe_qos = qos1
+              publish_qos = qos1
+              authentication = undefined
+
+              listeners.udp.default {
+                                     bind = 5683
+                                    }
+             }
+                        ">>).
+
+-define(LOGT(Format, Args), ct:pal("TEST_SUITE: " ++ Format, Args)).
+-define(PS_PREFIX, "coap://127.0.0.1/ps").
+-define(MQTT_PREFIX, "coap://127.0.0.1/mqtt").
+
+
+all() -> emqx_ct:all(?MODULE).
+
+init_per_suite(Config) ->
+    emqx_ct_helpers:start_apps([emqx_gateway], fun set_special_cfg/1),
+    Config.
+
+set_special_cfg(emqx_gateway) ->
+    ok = emqx_config:init_load(emqx_gateway_schema, ?CONF_DEFAULT);
+
+set_special_cfg(_) ->
+    ok.
+
+end_per_suite(Config) ->
+    emqx_ct_helpers:stop_apps([emqx_gateway]),
+    Config.
+
+%%--------------------------------------------------------------------
+%% Test Cases
+%%--------------------------------------------------------------------
+t_connection(_Config) ->
+    Action = fun(Channel) ->
```

Comment:
Is the indentation here on purpose or is it a problem with your automatic formatting tool?
Generally, we usually indent only 4 spaces, e.g.
```
Action = fun(Channel) ->
    …
end,

%% Or
Action = fun(Channel) ->
            …
         end,
```
We should not have any uniform norms now. Here is just a discussion

Reply:
sorry, this is a "feature" of erlang/otp emacs-mode,  I will find some time try to fix it